app/lib/search_query_parser.rb

```diff
@@ -1,15 +1,56 @@
 # frozen_string_literal: true
 
 class SearchQueryParser < Parslet::Parser
-  rule(:term)      { match('[^\s":]').repeat(1).as(:term) }
-  rule(:quote)     { str('"') }
+  # Efficiently matches disjoint strings
+  class StrList < Parslet::Atoms::Base
+    attr_reader :strings
+
+    def initialize(strings)
+      super()
+
+      @strings = strings
+      @pattern = Regexp.union(strings)
+      @min_length = strings.map(&:length).min
+    end
+
+    def error_msgs
+      @error_msgs ||= {
+        premature: 'Premature end of input',
+        failed: "Expected any of #{strings.inspect}, but got ",
+      }
+    end
+
+    def try(source, context, _consume_all)
+      match = source.match(@pattern)
+      return succ(source.consume(match)) unless match.nil?
+
+      # Input ending early:
+      return context.err(self, source, error_msgs[:premature]) if source.chars_left < @min_length
+
+      # Expected something, but got something else instead:
+      error_pos = source.pos
+      context.err_at(self, source, [error_msgs[:failed], source.consume(@len)], error_pos)
+    end
+
+    def to_s_inner(_prec)
+      "[#{strings.map { |str| "'#{str}'" }.join(',')}]"
+    end
+  end
+
+  rule(:term)      { match('[^\s]').repeat(1).as(:term) }
   rule(:colon)     { str(':') }
   rule(:space)     { match('\s').repeat(1) }
   rule(:operator)  { (str('+') | str('-')).as(:operator) }
-  rule(:prefix)    { term >> colon }
-  rule(:shortcode) { (colon >> term >> colon.maybe).as(:shortcode) }
-  rule(:phrase)    { (quote >> (term >> space.maybe).repeat >> quote).as(:phrase) }
-  rule(:clause)    { (operator.maybe >> prefix.maybe.as(:prefix) >> (phrase | term | shortcode)).as(:clause) | prefix.as(:clause) | quote.as(:junk) }
+  rule(:prefix_operator) { StrList.new(%w(has is language from before after during in)) }
```

Comment:
Is it worth making this list a bit easier to find in case things are ever added in the future?

Reply:
Yeah, I suppose this could be a constant at the top of the file, for instance!